internal/aws/k8s/k8sclient/daemonset_info.go

```diff
@@ -0,0 +1,28 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sclient // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/k8s/k8sclient"
+
+type DaemonSetInfo struct {
+	Name      string
+	Namespace string
+	Status    *DaemonSetStatus
+}
+
+type DaemonSetStatus struct {
+	NumberAvailable        int32
```

Comment:
mild nit - why are we specifying int32 here? Also, should it be a uint32 since it doesn't make sense for it to be negative (or am I misunderstanding)?

Reply:
Yeah thats fair.. I kept these the same as how they are modelled as the responses to the REST API in the k8s go client itself... but I cant imagine a scenario these would be -ve (in fact CW may drop negative metrics).

So if we think its better to convert these, I can force convert them to unit32.